src/helpers/validators/validateString/index.ts

```diff
@@ -0,0 +1,121 @@
+// import helpers
+import validateRegex from '../shared/helpers/validateRegex';
+import ValidationResult from '../shared/types/ValidationResult';
+
+// import types
+import StringValidationRequirements from './StringValidationRequirements';
+
+/**
+ * Determines whether a given input string is considered valid based on
+ *   the provided requirements.
+ * @author Austen Money
+ * @param input input string
+ * @param reqs requirements that the provided string should conform to
+ * @returns whether input is considered valid according to reqs - if
+ *   valid, returns a cleaned version of input; if invalid, returns
+ *   a string containing error messages describing which requirements
+ *   were not met.
+ */
+const validateString = (
+  input: string,
+  reqs: StringValidationRequirements,
+): ValidationResult => {
+  // stores all invalid input errors
+  const errorMessages: Array<string> = [];
+  // contains version of input that will be returned
+  let cleanedValue: string = input;
+
+  // remove whitespace if required
+  if (reqs.ignoreWhitespace) {
+    cleanedValue = input.replace(/\s+/g, '');
+  }
+
+  // apply max char requirement
+  if (reqs.minLen) {
+    if (cleanedValue.length < reqs.minLen) {
+      errorMessages.push(`Input must not be under ${reqs.minLen} character(s).`);
+    }
+  }
+
+  // apply max char requirement
+  if (reqs.maxLen) {
+    if (cleanedValue.length > reqs.maxLen) {
+      errorMessages.push(`Input must not be over ${reqs.maxLen} character(s).`);
+    }
+  }
+
+  // apply alphabetical requirement
+  if (reqs.lettersOnly) {
+    for (let i = 0, len = cleanedValue.length; i < len; i++) {
+      const curr = cleanedValue.charCodeAt(i);
+
+      // check that char is an upper or lower-case letter
+      if (!(curr > 64 && curr < 91) && 
+          !(curr > 96 && curr < 123)) {
+        errorMessages.push('Input must not contain non-letters.');
```

Comment:
Double negatives confuse me, so I would prefer something like `'Input must only contain letters'` - but either way works.

Reply:
Good point! I think that's a good call